utils/staking-rewards/snapshot.ts

```diff
@@ -0,0 +1,120 @@
+import {
+    LiquidityProtectionStore__factory as LiquidityProtectionStore,
+    StakingRewards__factory as StakingRewards
+} from '../../typechain';
+import { BigNumber, providers, utils } from 'ethers';
+import fs from 'fs';
+import MerkleTree from 'merkletreejs';
+
+const { solidityKeccak256, keccak256 } = utils;
+
+const ETHEREUM_PROVIDER_URL = 'ws://localhost:8545';
+const START_BLOCK = 11039642;
+const BATCH_SIZE = 1000;
+
+const LIQUIDITY_PROTECTION_STORE_ADDRESS = '0xf5fab5dbd2f3bf675de4cb76517d4767013cfb55';
+const STAKING_REWARDS_ADDRESS = '0x318fEA7e45A7D3aC5999DA7e1055F5982eEB3E67';
+
+let provider: providers.WebSocketProvider;
+
+const init = () => {
+    console.log(`Connecting to ${ETHEREUM_PROVIDER_URL}...`);
+
+    provider = new providers.WebSocketProvider(ETHEREUM_PROVIDER_URL);
+};
+
+const getProviders = async () => {
+    const toBlock = await provider.getBlockNumber();
+
+    console.log(`Getting all historic providers from ${START_BLOCK} to ${toBlock}...`);
+    console.log();
+
+    const store = LiquidityProtectionStore.connect(LIQUIDITY_PROTECTION_STORE_ADDRESS, provider);
+
+    const providers = new Set<string>();
+
+    for (let i = START_BLOCK; i < toBlock; i += BATCH_SIZE) {
```

Comment:
Semantic: I'd rename `i` to `startBlock`

Reply:
👍